### ISSUE TITLE
wrong line display after setOption()

### ISSUE THREAD

**Issue by tmtron** [apache/echarts#10228]
### Version
4.2.1

### Reproduction link
[https://jsfiddle.net/tmtron/brndo1hf/31/](https://jsfiddle.net/tmtron/brndo1hf/31/)

### Steps to reproduce
Open the jsfilddle example 
![image](https://user-images.githubusercontent.com/20422904/55670907-012d7c00-588a-11e9-9782-3c8af30e2adf.png)

and now press the `Update` button below the chart:
This will call `setOption()` with the `notMerge` parameter set to `true`.
The new data will show a wrong line, that is not in the data - see the yellow highlighted part (1):
![image](https://user-images.githubusercontent.com/20422904/55670914-2b7f3980-588a-11e9-8a3c-1ac4b1c7d8c0.png)
When you now move the zoom slider (2) to the left and back the data is displayed correctly
![image](https://user-images.githubusercontent.com/20422904/55670927-4d78bc00-588a-11e9-9163-38ddb7c2ce2e.png)

### What is expected?
Show the new data as it is.

### What is actually happening?
We see a wrong line in the new data.

---
notes: 
* the `notMerge` parameter of `setOption()` ist `true`, but still the wrong line is shown
* to make sure, that the data is correct, we can directly display `option2` in the js-fiddle example
  * replace `myChart.setOption(option);`
  * with `myChart.setOption(option2);`
  * then press the `Run` button at the left top --> the data is shown correctly
* maybe this is related to #6202
* as a workaround I can call `myChart.clear()` before setting the option
* it seems to be related to the `filterMode`:
  *  `filter` and `weakFilter`: show the wrong line
  *  `empty` and `none` work fine

**Comment by Ovilia** [comment]
Can you provide a demo with minimum data?

**Comment by tmtron** [comment]
I've removed lots of data and some more settings, so that the issue still occurs: https://jsfiddle.net/tmtron/brndo1hf/72/

I noticed something that may be helpful:
* the time of the first data-point in `datasetSource1` is before `xAxis.min`
* when we update this first datapoint to be `=> xAxis.min` the issue also does not occur
```
1554139666196 time of first value
1554139674662 xAxis min value 
```

**Comment by tmtron** [comment]
I hope the new [replace-merge](https://github.com/apache/incubator-echarts/pull/12987) mode in version 5.x can fix this

**Comment by tmtron** [comment]
Verified, that it's fixed in 5.1.2: see [jsfiddle](https://jsfiddle.net/tmtron/p1ayoLkc/5/)
